rmw_fastrtps_shared_cpp/src/namespace_prefix.cpp

```diff
@@ -31,8 +31,10 @@ const std::vector<std::string> _ros_prefixes =
 std::string
 _resolve_prefix(const std::string & name, const std::string & prefix)
 {
-  if (name.rfind(prefix, 0) == 0 && name.at(prefix.length()) == '/') {
-    return name.substr(prefix.length());
+  if (name.length() > prefix.length()) {
```

Comment:
Coming back with fresh eyes on this one, I think this can all be simplified down to:
```
if (name.rfind(prefix + "/", 0)) {
  return name.substr(prefix.length());
}
```

Same thing with the functions below.

Reply:
That is correct. Fixed in https://github.com/ros2/rmw_fastrtps/pull/388/commits/e0fe9ca02c9d9287f990f7855f87b957e6470f48.